sdk/lib/drivers/sound/mmixer/mixer.c

```diff
@@ -836,10 +836,9 @@ MMixerInitialize(
             }
             else
             {
-                DPRINT1("Failed to enumerate device %lu\n", DeviceIndex);
+                DPRINT1("Failed to enumerate device %lu, Status %x\n", DeviceIndex, Status);
 
-                /* TODO cleanup */
-                return Status;
+                /* ignore error and continue */
```

Comment:
Should this be done also when `MMixerCreateMixerData()` fails?
If not, could you add a comment there to explain why not?

Reply:
> Should this be done also when MMixerCreateMixerData() fails?

Indeed. If another part failed, we also can skip the failure and continue with the next device then. It also looks correct. Just breaking out from the loop is definitely not correct. However the possibility of `MMixerCreateMixerData()` failure is much less because it only does an allocation and can fail only on this. Otherwise it returns success for all other cases.

Reply:
Done, thanks.